examples/nextjs/middleware.ts

```diff
@@ -0,0 +1,55 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { SDK, Config } from '@corbado/node-sdk';
+import { jwtDecode } from 'jwt-decode';
+
+function readUserIP(request: NextRequest): string {
+  return '127.0.0.1';
+  // return (
+  //   request.ip ??
+  //   request.headers.get('x-real-ip') ??
+  //   request.headers.get('x-forwarded-for') ??
+  //   ''
+  // );
+}
+
+function validateSession(shortSession: string | undefined) {
+  if (!shortSession) {
+    return false;
+  }
+
+  const decodedShortSession = jwtDecode(shortSession);
+  return !!decodedShortSession.exp && decodedShortSession.exp > Date.now() / 1000;
```

Comment:
We need to do a real JWT validation here, just decoding the JWT is not enough. Verification can be done either with validateShortSessionValue or with getCurrentUser from the NodeSDK.